File: application-configuration/src/main/java/org/acme/config/GreetingResource.java
package org.acme.config;

import java.util.Optional;

import javax.inject.Inject;
import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

import org.eclipse.microprofile.config.inject.ConfigProperty;

@Path("/greeting")
public class GreetingResource {

    @Inject
    @ConfigProperty(name = "greeting.message")
    private String message;

    @Inject
    @ConfigProperty(name = "greeting.suffix", defaultValue="!")
    private String suffix;

    @Inject
    @ConfigProperty(name = "greeting.name")
    private Optional<String> name;


    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public String hello() {
        return message + " " + name.orElse("world") + suffix;
    }
}

File: application-configuration-kotlin/src/main/kotlin/org/acme/config/GreetingResource.kt
package org.acme.config

import org.eclipse.microprofile.config.inject.ConfigProperty
import javax.ws.rs.GET
import javax.ws.rs.Path
import javax.ws.rs.Produces
import javax.ws.rs.core.MediaType

@Path("/greeting")
class GreetingResource {
    @ConfigProperty(name = "greeting.message")
    lateinit var message: String
    @ConfigProperty(name = "greeting.suffix", defaultValue = "!")
    lateinit var suffix: String
    @ConfigProperty(name = "greeting.name")
    var name: String? = null

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    fun hello(): String {
        return "$message ${name ?: "world"}$suffix"
    }
}